gabriel-portfolio/portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,42 @@
 package com.google.sps.servlets;
 
 import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import com.google.gson.Gson;
 import javax.servlet.annotation.WebServlet;
 import javax.servlet.http.HttpServlet;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** Servlet that returns some example content. TODO: modify this file to handle comments data */
-// Did not remove this because it will be used in week 3
-// @WebServlet("/data")
-// public class DataServlet extends HttpServlet {
-//   @Override
-//   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-//     response.setContentType("text/html;");
-//     response.getWriter().println("<h1>Hello world!</h1>");
-//   }
-// }
+
+@WebServlet("/data")
+public class DataServlet extends HttpServlet {
+  
+  private List<String> messages;
+
+  @Override
+  public void init() {
+    messages = new ArrayList<>();
+    messages.add("What do you YOU");
+    messages.add("care about what");
+    messages.add("other people think");
+  }
+  
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    response.setContentType("application/json;");
+    String json = convertToJsonUsingGson(messages);
+    response.getWriter().println(json);
+  }
+
+  /*
+    Converts an arraylist to an JSON string using the Gson library
+  */
+  private String convertToJsonUsingGson(List<String> list) {
```

Comment:
no need to have a separate method unless additional logic other than a call to  toJson() method is added

Reply:
Resolved